scalableminds/webknossos
# Separate DataStore and TracingStore

### URL of deployed dev instance (used for testing):
- https://___.webknossos.xyz

### Notes
- note that tracing related api routes no longer have the `data/` prefix
- note that you might have to refresh your schema so that initialData is inserted again (with tracingStore info)

### TODO:
 - [x] adapt frontend to use tracingstore from annotation info
 - [x] offer route to get volume tracing data in tracing store → `tracings/volume/:tracingId/data` expecting POST JSON body just like binary data loading
 - [x] adapt frontend to that
 - [x] test as standalone
 - [x] adapt docker files
 - [x] remove fossildb from datastore
 - [x] check which config settings are now unused
 - [x] remove obsolete `api/tracingstore`
 - [x] when initializing volume, set boundingbox, elementclass and largestSegmentId already on wK side
 - [x] trigger tracingstore health check from frontend?
 - [x] adapt the e2e tests to new volumetracing handling
 - [ ] is the dataStore field in annotation info api now unused in the frontend? if so, remove it
 - [x] how do we publish tracingstore to dockerhub?

### Steps to test:
- trace some, both in volume and skeleton
- set up tracingStore as standalone app [(→setup)](https://github.com/scalableminds/webknossos/wiki/Set-up-a-standalone-tracingstore-locally-(preliminary,-not-in-master-yet)), should still be able to trace both with skeleton + volume
- download + reupload volume tracing

### Issues:
- fixes #3072 

------
